IRuettae/Core/IRuettae.Core.LocalSolver/Solver.cs

```diff
@@ -0,0 +1,418 @@
+
+
+                        // visiting
+                        var visitDurationSelector = model.Function(i => visitDurationArray[sequence[i]]);
+                        santaVisitDurations[s] = model.Sum(model.Range(0, c), visitDurationSelector);
+
+                        // time slot
+                        var currDayIndex = day; // copy because used in lambda expression
+
+                        var waitBetweenVisits = UseWaitBetweenVisits ? model.Array(santaWaitBetweenVisit[s]) : model.Int(0,0);
+
+                        var visitStartingTimeSelector = model.Function((i, prev) =>
+                            model.If(i == 0,
+                                input.Days[currDayIndex].from + santaWaitBeforeStart[s] + distanceFromHomeArray[sequence[i]],
+                                prev + visitDurationArray[sequence[i - 1]] + distanceArray[sequence[i - 1], sequence[i]] + (UseWaitBetweenVisits ? waitBetweenVisits[sequence[i]] : model.Int(0, 0))
+                            )
+                        );
+
+                        var visitStartingTime = model.Array(model.Range(0, c), visitStartingTimeSelector);
+                        santaVisitStartingTimes[s] = visitStartingTime;
+
+                        // desired
+                        var visitDesiredDurationSelector = model.Function(i =>
+                        {
+                            var v = sequence[i];
+                            var nDesired = visitDesiredCountArray[v];
+
+                            var visitStart = visitStartingTime[i];
+                            var visitEnd = visitStart + visitDurationArray[sequence[i]];
+
+                            var desiredIntersection = model.Function((n) =>
+                            {
+                                // desired start
+                                var x = model.If(nDesired == 0, model.Int(0, 0),
+                                        model.At(visitDesiredArray, v, n, model.Int(0, 0)));
+                                // desired end
+                                var y = model.If(nDesired == 0, model.Int(0, 0),
+                                        model.At(visitDesiredArray, v, n, model.Int(1, 1)));
+
+                                return model.If(model.Or(y < visitStart, x > visitEnd),
+                                    // if no intersection    
+                                    0,
+                                    //else
+                                    model.Min(y, visitEnd) - model.Max(x, visitStart)
+                                );
+                            });
+                            return model.Sum(model.Range(0, nDesired), desiredIntersection);
+                        });
+                        santaDesiredDuration[s] = model.Sum(model.Range(0, c), visitDesiredDurationSelector);
+
+
+                        // unavailable
+                        var visitUnavailableDurationSelector = model.Function(i =>
```

Comment:
Ev. Code von Desired reusen.

Reply:
Wenn du mir sagst wie ich die arrays in diese LocalSolver Funktion kriege, gerne ;)